Move peer handshake resets out of UpdatePeers and into new method Res…

Rizzly opened the pull request:
…etPeers

This should reduce the time UpdatePeers needs to run, and allow us to reset handshake more often.

Marking as draft since this still has some `TODO`:s that need investigating/fixing, some tests still need to be performed and we might want to refactor out connection reporting as part of this PR as well.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mullvad/wg-manager/24)
<!-- Reviewable:end -->

gdetrez commented on wireguard/wireguard.go:
```diff
@@ -42,6 +42,69 @@ func New(interfaces []string, metrics *statsd.Client) (*Wireguard, error) {
+			continue
+		}
+
+		// Add new peers, remove deleted peers, and remove peers should be reset
```
Nitpick: update this comment and the one it was copied from down bellow

Rizzly replied:
Nice catch, fixed!

gdetrez reviewed (approved):
I think we accidentally introduced some comment drift :speak_no_evil:
Looks good otherwise!

jbjorkang reviewed (commented):
Looks good to me, as far as I understand!

Rizzly commented:
I've now tested this quite extensively in our staging environment, added missing parts and fixed a few bugs. I've also refactored the connection reporting as talked about. Still missing is clarity on what to do with the preshared keys, I'm awaiting feedback there from other teams. Might be that we can simply remove it.

gdetrez reviewed (approved):
LGTM!

oskaralmlov reviewed (approved):
Looks good for my limited understanding of goalang :rocket:
